fe/src/main/java/org/apache/doris/task/DeleteJob.java

```diff
@@ -0,0 +1,185 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.task;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.Replica;
+import org.apache.doris.common.Config;
+import org.apache.doris.common.MetaNotFoundException;
+import org.apache.doris.load.DeleteInfo;
+import org.apache.doris.load.TabletDeleteInfo;
+import org.apache.doris.transaction.AbstractTxnStateChangeCallback;
+import org.apache.doris.transaction.TransactionState;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+public class DeleteJob extends AbstractTxnStateChangeCallback {
+    private static final Logger LOG = LogManager.getLogger(DeleteJob.class);
+
+    public enum DeleteState {
+        UN_QUORUM,
+        QUORUM_FINISHED,
+        FINISHED
+    }
+
+    private DeleteState state;
+
+    // jobId(listenerId). use in beginTransaction to callback function
+    private long id;
+    // transaction id.
+    private long signature;
+    private Set<Long> totalTablets;
+    private Set<Long> quorumTablets;
+    private Set<Long> finishedTablets;
+    Map<Long, TabletDeleteInfo> tabletDeleteInfoMap;
```

Comment:
There will be concurrent issue when visiting `tabletDeleteInfoMap` from different thread: the report task thread and DeleteHandler thread. So you should use ConcurrentHashMap